core/tee/tee_svc_storage.c

```diff
@@ -1322,9 +1338,34 @@ TEE_Result syscall_storage_obj_write(unsigned long obj, void *data, size_t len)
 		goto exit;
 
 	o->info.dataPosition += len;
-	if (o->info.dataPosition > o->info.dataSize)
+	if (o->info.dataPosition > o->info.dataSize) {
+		struct tee_svc_storage_head *head = NULL;
+		tee_fs_off_t off;
+
 		o->info.dataSize = o->info.dataPosition;
 
+		/* Get current offset */
+		off = o->pobj->fops->lseek(&res, o->fd, 0, TEE_FS_SEEK_CUR);
+
+		head = o->pobj->head;
+
+		if (head && head->magic == TEE_SVC_STORAGE_MAGIC) {
```

Comment:
Yes I think this should cause the TA to panic, what would be the right thing to do here?